src/main/scala/com/fulcrumgenomics/util/Metric.scala

```diff
@@ -101,14 +102,30 @@ object Metric {
   /** Reads metrics from a set of lines.  The first line should be the header with the field names.  Each subsequent
     * line should be a single metric. */
   def iterator[T <: Metric](lines: Iterator[String], source: Option[String] = None)(implicit tt: ru.TypeTag[T]): Iterator[T] = {
-    if (lines.isEmpty) throw new IllegalArgumentException(s"No header found in metrics" + source.map(" in source: " + _).getOrElse(""))
+    val clazz: Class[T]   = ReflectionUtil.typeTagToClass[T]
+
+    def fail(lineNumber: Int,
```

Comment:
Why not just have this be a `private def` at the top level inside `Metric` rather than nested inside this other method?

Reply:
I had it that way, but this way it makes calling the method much more concise in-line below.   It's not too much overhead since it's outside the main loop.